wpilibcExamples/src/main/cpp/examples/StateSpaceElevator/cpp/Robot.cpp

```diff
@@ -0,0 +1,163 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2017-2020 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include <frc/Encoder.h>
+#include <frc/GenericHID.h>
+#include <frc/PWMVictorSPX.h>
+#include <frc/StateSpaceUtil.h>
+#include <frc/TimedRobot.h>
+#include <frc/XboxController.h>
+#include <frc/controller/LinearQuadraticRegulator.h>
+#include <frc/drive/DifferentialDrive.h>
+#include <frc/estimator/KalmanFilter.h>
+#include <frc/system/LinearSystemLoop.h>
+#include <frc/system/plant/DCMotor.h>
+#include <frc/system/plant/LinearSystemId.h>
+#include <frc/trajectory/TrapezoidProfile.h>
+#include <units/acceleration.h>
+#include <units/length.h>
+#include <units/mass.h>
+#include <units/velocity.h>
+#include <wpi/math>
+
+/**
+ * This is a sample program to demonstrate how to use a state-space controller
+ * to control an elevator.
+ */
+class Robot : public frc::TimedRobot {
+  const int kMotorPort = 0;
+  const int kEncoderAChannel = 0;
+  const int kEncoderBChannel = 1;
+  const int kJoystickPort = 0;
+
+  const units::meter_t kRaisedPosition = 2_ft;
+  const units::meter_t kLoweredPosition = 0_ft;
+
+  const units::meter_t kDrumRadius = 0.75_in;
+
+  /*
+  The plant holds a state-space model of our flywheel. In this system the states
+  are as follows: States: [velocity], in RPM. Inputs (what we can "put in"):
+  [voltage], in volts. Outputs (what we can measure): [velocity], in RPM.
+   */
+  frc::LinearSystem<2, 1, 1> m_elevatorPlant = [this] {
+    auto motors = frc::DCMotor::NEO(2);
+
+    // carriage mass
+    auto m = 4.5_kg;
+
+    // reduction between motors and encoder,
+    // as output over input. If the elevator spins slower than the motors, this
+    // number should be greater than one.
+    auto G = 6.0;
+
+    return frc::LinearSystemId::ElevatorSystem(motors, m, kDrumRadius, G);
+  }();
```

Comment:
Make constants in the class for things in here, then do the following.
```suggestion
  frc::LinearSystem<2, 1, 1> m_elevatorPlant{
    frc::DCMotor::NEO(2), kCarriageMass, kDrumRadius, kGearRatio};
```